lisp/ess-r-mode.el

```diff
@@ -917,6 +921,7 @@ Examples: (ess-current-R-at-least '2.7.0)
 (defun ess-newest-r (rvers)
```

Comment:
The  semantics of names `ess-find-newest-R` and `ess-newest-r` is unfortunate. I thought that the latter is actually running the newest R.  Shall we make the latter internal and name it more specifically, like `ess-r--newest-r-path`? Or just inline it whenever needed; it's a 2 liner. Capital R in one and not in the other is confusing.

Reply:
Could be, but it then becomes semantically similar to `inferior-ess-r-program` which is not desirable.

Both `ess-newest-r` and the inner `ess-find-newest-date` are used once and have bad names. I would suggest we make a local `cl-labels` inside `ess-find-newest-R-version` and remove both of the above. Or at least inline `ess-find-newest-date` into `ess-newest-r` and rename the latter into an internal with a more suggestive name.